cmd/minikube/cmd/config/config_test.go

```diff
@@ -41,12 +41,14 @@ var configTestCases = []configTestCase{
     "ReminderWaitPeriodInHours": 99,
     "cpus": 4,
     "disk-size": "20g",
+    "iso-url": "http://foo.bar/minishift-centos.iso",
     "log_dir": "/etc/hosts",
     "show-libmachine-logs": true,
     "v": 5,
     "vm-driver": "kvm"
 }`,
 		config: map[string]interface{}{
+			"iso-url": "http://foo.bar/minishift-centos.iso",
```

Comment:
> I'd like to now whether we can add this to config.json or whether we need a different config file.

I would say we should use a different config file and add some intelligent there to verify key-value pair which and that config can be modified by manually. Currently config.json file used by viper and I am not sure if we can add some magic there to validate what key and respective value and that's why `set` and `get` make more sense here to modify that file instead manually where validator functions are already in place.

> So if I manually add another flag (which is not listed here) to the config, I think it will be picked up fine by Viper. I think there might be some things we need to think through here.

Yes if you add manually then Viper will pick it up that's how viper works, it just read that config file  https://github.com/minishift/minishift/blob/master/cmd/minikube/cmd/root.go#L129-L138 

> What's about config writes? Say I have additional data and read the config, modify it and then save it again (using minishift get/set), does the additional data persist?

Yes, I added a set test for that it will update the key value and then make it persistent.